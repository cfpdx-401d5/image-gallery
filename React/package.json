{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
