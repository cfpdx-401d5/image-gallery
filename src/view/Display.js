import React from 'react';
import EditForm from '../forms/EditForm';

function ChooseView(props) {
    switch (props.view) {

        case 'gallery':
            return (
                <div>
                    <p><b>Title:</b> {props.item.title}</p>
                    <img className='gallery' src={props.item.url} alt='cute bunnys'/>
                    <p><b>Description:</b> {props.item.description}</p>
                </div>
            );

        case 'details':
            return (
                <div>
                    <p><b>Title:</b> {props.item.title}</p>
                    <p><b>Description:</b> {props.item.description}</p>
                    <p><b>URL:</b> <a href={props.item.url}>{props.item.url}</a></p>
                </div>
            );

        case 'thumbnail':
            return (
                <div>
                    <p><b>Title:</b> {props.item.title}</p>
                    <img className='thumbnail' src={props.item.url} alt='cute bunnys'/>
                </div>
            );

        default:
            return (
                'Something went wrong.'
            );
    }
}

function Buttons(props) {
    return (
        <div>
            <button className='buttons' onClick={() => props.onDelete(props.item.id)}>Delete</button>
            <button className='buttons' onClick={() => props.edit()}>Edit</button>
            <div>{props.isEditable && <EditForm item={props.item} onChange={props.onChange} onDisplay={props.edit} onEdit={props.onEdit} />}</div>
        </div>
    );
}

class Display extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isEditable: false
        };
        this.editForm = this.editForm.bind(this);
    }

    editForm() {
        this.setState({
            isEditable: !this.state.isEditable
        });
    }

    render() {
        return (
            <li>
                <ChooseView 
                    item={this.props.item} 
                    view={this.props.view} />

                <Buttons 
                    edit={this.editForm} 
                    isEditable={this.state.isEditable}
                    onDelete={this.props.onDelete} 
                    onChange={this.props.onChange} 
                    onEdit={this.props.onEdit}
                    item={this.props.item} />
                    <hr />
            </li>
        );
    }
}

function DisplayList(props) {
    const listItem = props.array.map((item, index) => {
        return (
            <Display 
            key={index} 
            item={item} 
            onDelete={props.onDelete} 
            onChange={props.onChange} 
            onEdit={props.onEdit} 
            view={props.view} />
        );
    });
    return (
      <ul>{listItem}</ul>
    );
}

export { DisplayList };

DisplayList.propTypes = {
    array: React.PropTypes.array,
    view: React.PropTypes.string
};

Display.propTypes = {
    onDelete: React.PropTypes.func,
    onEdit: React.PropTypes.func,
    onChange: React.PropTypes.func,
    item: React.PropTypes.object,
    view: React.PropTypes.string
};

Buttons.propTypes = {
    onDelete: React.PropTypes.func,
    onEdit: React.PropTypes.func,
    onChange: React.PropTypes.func,
    item: React.PropTypes.object,
    editForm: React.PropTypes.func,
    isEditable: React.PropTypes.bool,
    edit: React.PropTypes.func
};

ChooseView.propTypes = {
    view: React.PropTypes.string,
    item: React.PropTypes.object
};
