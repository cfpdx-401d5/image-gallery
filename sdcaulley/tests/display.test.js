/* eslint-disable */
import React from 'react';
import {shallow, render} from 'enzyme';
import toJson from 'enzyme-to-json';
import Display from '../src/view/display-view';

describe('list display', () => {
    it('list display', () => {
        const rendered = shallow(<Display array={[]} />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});