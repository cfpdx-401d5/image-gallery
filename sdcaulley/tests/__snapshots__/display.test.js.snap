exports[`list display list display 1`] = `
<div>
  <ViewSelector />
  <h3>
    Click a button to view bunnies!
  </h3>
</div>
`;
